File: doc/codes/python/examples/api/compute/pattern/plane-example.py
import numpy as np
from autd3.geometry import Autd3, Geometry
from autd3.units import m, s
from autd3_pattern import plane, wavelength

geometry = Geometry([Autd3([0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0])])

phases = geometry.phase_buffer()

plane(
    geometry,
    np.array([0.0, 0.0, 1.0]),
    wavelength(340 * m / s),
    phases,
)

File: doc/codes/rust/examples/api/compute/pattern/plane-example.rs
use autd3_rs::geometry::{Autd3, Geometry, Vector3};
use autd3_rs::units::{m, s};
use autd3_rs_pattern::{plane, wavelength};

// HIDE
fn main() {
    // HIDE_END
    let geometry = Geometry::new(vec![Autd3::default()]);

    let mut phases = geometry.phase_buffer();

    plane(
        &geometry,
        Vector3::z_axis(),
        wavelength(340.0 * m / s),
        &mut phases,
    );
    // HIDE
}
// HIDE_END
